core/frontend/src/render/webgl/CachedGeometry.ts

```diff
@@ -520,6 +520,14 @@ export class TexturedViewportQuadGeometry extends ViewportQuadGeometry {
     // There is no such 'type' as WebGLTexture at run-time.
     assert(this._textures.every((tx) => !(tx instanceof TextureHandle)));
   }
+
+  public static createTexturedViewportQuadGeometry(techniqueId: TechniqueId, textures: WebGLTexture[]): TexturedViewportQuadGeometry | undefined {
+    const params = ViewportQuad.getInstance().createParams();
+    if (undefined === params)
+      return undefined;
+
+    return new this(params, techniqueId, textures);

```

Comment:
is it our preference to use `this` over the explicit class name in static methods?

Reply:
`create` is defined on ViewportQuadGeometry but calling it on `ScreenSpaceGeometry` and using `new this` instantiates the subtype.
```
    const effectGeometry = ScreenSpaceGeometry.create(TechniqueId.Invalid);
    assert(effectGeometry instanceof ScreenSpaceGeometry);
```